src/adapters/pubgears.js

```diff
@@ -0,0 +1,194 @@
+var bidfactory = require('../bidfactory.js');
+var bidmanager = require('../bidmanager.js');
+var consts = require('../constants.json');
+var utils = require('../utils.js');
+var d = document;
+var SCRIPT = 'script';
+var PARAMS = 'params';
+var SIZES = 'sizes';
+var SIZE = 'size';
+var CPM = 'cpm';
+var AD = 'ad';
+var WIDTH = 'width';
+var HEIGHT = 'height';
+var PUB_ZONE = 'pub_zone';
+var GROSS_PRICE = 'gross_price';
+var RESOURCE = 'resource';
+var DETAIL = 'detail';
+var BIDDER_CODE_RESPONSE_KEY = 'bidderCode';
+
+var BIDDER_CODE = 'pubgears';
+
+var SCRIPT_ID = 'pg-header-tag';
+
+var ATTRIBUTE_PREFIX = 'data-bsm-';
+var SLOT_LIST_ATTRIBUTE = 'slot-list';
+var PUBLISHER_ATTRIBUTE = 'pub';
+var FLAG_ATTRIBUTE = 'flag';
+
+var PLACEMENT_CODE = 'placementCode';
+var BID_ID = 'bidId';
+var PUBLISHER_PARAM = 'publisherName';
+var PUB_ZONE_PARAM = 'pubZone';
+
+
+
+
+var BID_RECEIVED_EVENT_NAME = 'onBidResponse';
+var SLOT_READY_EVENT_NAME = 'onResourceComplete';
+
+var CREATIVE_TEMPLATE = decodeURIComponent("%3Cscript%3E%0A(function(define)%7B%0Adefine(function(a)%7B%0A%09var%20id%3D%20%22pg-ad-%22%20%2B%20Math.floor(Math.random()%20*%201e10)%2C%20d%3D%20document%0A%09d.write(\'%3Cdiv%20id%3D%22\'%2Bid%2B\'%22%3E%3C%2Fdiv%3E\')%0A%09a.push(%7B%0A%09%09pub%3A%20\'%25%25PUBLISHER_NAME%25%25\'%2C%0A%09%09pub_zone%3A%20\'%25%25PUB_ZONE%25%25\'%2C%0A%09%09sizes%3A%20%5B\'%25%25SIZE%25%25\'%5D%2C%0A%09%09flag%3A%20true%2C%0A%09%09container%3A%20d.getElementById(id)%2C%0A%09%7D)%3B%0A%7D)%7D)(function(f)%7Bvar%20key%3D\'uber_imps\'%2Ca%3Dthis%5Bkey%5D%3Dthis%5Bkey%5D%7C%7C%5B%5D%3Bf(a)%3B%7D)%3B%0A%3C%2Fscript%3E%0A%3Cscript%20src%3D%22%2F%2Fc.pubgears.com%2Ftags%2Fb%22%3E%3C%2Fscript%3E%0A");
+
+
+var TAG_URL = '//c.pubgears.com/tags/h';
+var publisher = '';
+
+module.exports = PubGearsAdapter;
+
+function PubGearsAdapter() {
+
+  var proxy = null;
+  var pendingSlots = {};
+  var initialized = false;
+
+  this.callBids = callBids;
+
+  function callBids(params) {
+
+    var bids = params[consts.JSON_MAPPING.PL_BIDS];
+    var slots = bids.map(getSlotFromBidParam) ;
+    if(slots.length <= 0)
+      return;
+    publisher = bids[0][PARAMS][PUBLISHER_PARAM];
+
+    bids.forEach(function(bid) {
+
+      var name = getSlotFromBidParam(bid);
+      pendingSlots[ name ] = bid;
+    });
+
+    proxy = proxy || getScript(SCRIPT_ID) || makeScript(slots, publisher, SCRIPT_ID, TAG_URL);
+    if(!initialized)
+        registerEventListeners(proxy);
+    initialized = true;
+  }
+
+  function loadScript(script) {
```

Comment:
Our API uses DOM events instead of method calls. The adapter registers event listeners with the script element [here](https://github.com/MbidIO/Prebid.js/blob/ebc6db3272dd5124fbcddc36d5684db0c1ce21bb/src/adapters/pubgears.js#L116-L117)
 
[API Documentation](http://c.pubgears.com/eg/nucleus/api/index.html)